### DISCUSSION TITLE
socket leak to googleusercontent.com

### DISCUSSION THREAD

**Discussion by ninja-** [KRTirtho/spotube#1259] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

lut 22 18:22:56 x com.github.KRTirtho.Spotube.desktop[1518019]: flutter: Error saving preferences to disk: FileSystemException: Cannot open file, path = '/home/user2/.var/app/com.github.KRTirtho.Spotube/data/spotube/shared_preferences.json' (OS Error: Too many open files, errno = 24)

➜  applications ls -lash /proc/1518019/fd -lash | wc -l
1016
➜  applications ls -lash /proc/1518019/fd -lash | grep socket | wc -l 
937

spotube 1518019 user2 1020u  IPv4 18433344      0t0  TCP x:51908->25.224.186.35.bc.googleusercontent.com:https (ESTABLISHED)
spotube 1518019 user2 1021u  IPv4 18433346      0t0  TCP x:51910->25.224.186.35.bc.googleusercontent.com:https (ESTABLISHED)
spotube 1518019 user2 1022u  IPv4 18433348      0t0  TCP x:51928->25.224.186.35.bc.googleusercontent.com:https (ESTABLISHED)
spotube 1518019 user2 1023u  IPv4 18433350      0t0  TCP x:51932->25.224.186.35.bc.googleusercontent.com:https (ESTABLISHED)
(....repeats 900 times...)

at the same time as the error, the application froze and was stuck on "application not responsive" screen from GNOME - unforunately gdb strackace I got, after decoding, doesn't reveal why - but maybe some internals were trying to establish a new file descriptor in a loop and were failing

### Expected Behavior

No socket leaks after using the app for long time

### Steps to reproduce

Run the FlatPak app for a while, skip a lot of songs

### Operating System

Ubuntu 23.10, latest flatpak

### Installation source

Flathub (Flatpak)

### Self grab

- [ ] I'm ready to work on this issue!

**Comment by KRTirtho** [comment]
You're using YoutTube audio source so of course it's gonna make calls to googleusercontent.com. Because the audio bytes of the stream are fetched from a GCP bucket or something similar internal backend solution that's used by YoutTube.

**Comment by ninja-** [comment]
as I mentioned, the application crashed exceeding 1024 open file descriptors. over 900 of them were connections to googleusercontent.com while the player was idle, having skipped through many songs before. it means there is a bug because connection to googleusercontent.com is not closed after song is changed or in some other funcionality like search engine
